common/testing/debugtimeout/debugtimeout.go

```diff
@@ -0,0 +1,26 @@
+//go:build test_dep || integration
+
+package debugtimeout
+
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+var Multiplier = getMultiplier()
+
+func getMultiplier() time.Duration {
+	v, ok := os.LookupEnv("TEMPORAL_DEBUG_TIMEOUT")
+	if !ok {
+		return time.Duration(1)
+	}
+	if v == "" {
+		return time.Duration(10)
+	}
+	parsed, err := strconv.ParseInt(v, 10, 64)
+	if err != nil {
+		panic("TEMPORAL_DEBUG_TIMEOUT should not be empty or a valid integer, got: " + v)
+	}
+	return time.Duration(parsed)
+}
```

Comment:
I changed the behavior to only use `10` when the env var is specified; and stay at `1` otherwise. Our functional tests were actually failing when it was set to `10`.

Also, I changed it to panic since this is only used in tests and can inform the developer that they're using this wrong.

Reply:
I don't think this is a good decision.  The version in ST results in 10 for all tests. Your approach means fixing tests one at a time to set the ENV var.  

I suspect your functional tests are failing because of what I said earlier: they are using the multiplier *in the tests*.  They should not.  In ST we only use the multiplier to affect "production" timeouts.

also, your scheme is more complicated ... not seeing the value.  Though I like the panic

Reply:
I was surprised by this comment so I went back to the original PR descriptions and realized that these two mechanisms seem to look the same at first but have different goals.

Internal:
> It should resolve a large number of CI **test flakes** (and provide a mechanism for fixing others) [...].

OSS:
> Default RPC timeouts work great for production environments. While in server **debugging mode** it is almost impossible to step over the code with default values.

That would explain our differences regarding the design (and your comments about using it in tests).